src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/SemanticModel/SharedInfo/IsActiveMethodInfo.cs

```diff
@@ -0,0 +1,47 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Acuminator.Utilities.Roslyn.Constants;
+
+using Microsoft.CodeAnalysis;
+
+namespace Acuminator.Utilities.Roslyn.Semantic.SharedInfo
+{
+	/// <summary>
+	/// Information about the IsActive special method in graph and DAC extensions.
+	/// </summary>
+	public class IsActiveMethodInfo : SymbolItem<IMethodSymbol>
+	{
+		public IsActiveMethodInfo(IMethodSymbol isActiveMethod) :
+							 base(isActiveMethod, declarationOrder: 0)
+		{
+		}
+
+		/// <summary>
+		/// Collects info about IsActive method from DAC or graph extension symbol and creates <see cref="IsActiveMethodInfo"/> DTO.
+		/// </summary>
+		/// <param name="dacOrGraphExtension">The DAC or graph extension.</param>
+		/// <returns>
+		/// The <see cref="IsActiveMethodInfo"/> DTO if extension contains IsActive method, otherwise <see langword="null"/>.
+		/// </returns>
+		internal static IsActiveMethodInfo? GetIsActiveMethodInfo(INamedTypeSymbol dacOrGraphExtension)
+		{
+			ImmutableArray<ISymbol> isActiveCandidates = dacOrGraphExtension.GetMembers(DelegateNames.IsActive);
+
+			if (isActiveCandidates.IsDefaultOrEmpty)
+				return null;
+
+			IMethodSymbol isActiveMethod =
+				isActiveCandidates.OfType<IMethodSymbol>()
+								  .FirstOrDefault(method => method.IsStatic && method.DeclaredAccessibility == Accessibility.Public &&
+															method.Parameters.IsDefaultOrEmpty && !method.IsGenericMethod &&
```

Comment:
I agree. The question is even broader than this: maybe we'd like to remove this API from the Acumatica Framework itself since it is not used anywhere.

Reply:
I think we shouldn't. It is meant for graph extensions especially in cases where developer doesn't control original graph code. 

The use case are class hierarchies of graphs. Usually graph extension for a base graph is applied to all derived graphs as well. This method allows to disable extension for some of these derived graphs (or for a base graph) and it covers a small spot not covered by the standard `IsActive` method. So it's not some duplicating functionality. It does something that other APIs don't.

We don't have many derived graphs in the application code (I can think only of `APInvoiceEntry`/`SOInvoiceEntry`). I think this is the reason why this API is not encountered in our code. But our partners sometimes make a copy of a screen with a bit different UI by creating a derived graph. Also it is possible that we will need such API for ourselves in the future.
If I understand correctly the feature was requested by the Services team that may also mean that it is more useful for external developers.